writeup/driftsort_introduction/text.md

```diff
@@ -40,92 +42,146 @@ The primary goal was to develop a replacement for the current Rust standard libr
-- **Mergesort**: Mergesort fallback used when recursion depth limit is reached in the Quicksort. Uses driftsort with Quicksort run creation disabled, always eagerly creating runs with the small-sort if none are found.
+- **Merge**: Branchless merge derived from the current `slice::sort` implementation. Requires at most N / 2 auxiliary memory.
+- **Sorted run detection**: Linear scan with early exit to find existing runs of sorted elements in the input. In contrast to glidesort which uses a fixed length after which it counts an existing run as already sorted, driftsort has a min run length of ~sqrt(N). In addition driftsort will not try to find another run within the min run length region, if the first attempt failed. The advantage of disrupting the top-down Quicksort process to allow merging with an existing run, requires a certain length relative to N to be worth it. This approach also avoids trying to find and merge small streaks in inputs with few unique elements, which are more efficiently handled by the top-down Quicksort.
+- **Quicksort**: Top-down partitioning loop and algorithm.
+- **Pivot selection**: Recursive median-of-three approximating sqrt(N) sampling.
+- **Hybrid small-sorts**: Two different small-sort implementations as a base case for recursive quicksort, chosen at compile time based on type characteristics. Insertion sort is used as a fallback for non `Freeze` types. A novel hybrid stable small-sort, based on a fast fixed function for sorting 4 elements combined with bidirectional merging and insertion sort is used for `Freeze` types, shared with ipnsort. This small-sort approach achieves comparable results with the one in glidesort, but does so at a significantly smaller binary-size and compile-time cost.
+- **Ancestor pivot tracking**: By keeping track of earlier used pivots in Quicksort we can detect and process common elements, allowing for O(N * log(K)) total comparisons where K is the number of distinct elements. This is also great for Zipfian distributions.
+- **Partition**: Branchless stable partition. Left-to-right scan that performs a `cmov` style pointer select to fill the auxiliary memory buffer from left-to-right or right-to-left depending on the comparison outcome.
+- **Mergesort**: Mergesort fallback used when a recursion depth limit is reached in the Quicksort. Re-uses the existing merging code in driftsort by calling itself with a flag that forces small runs to be made with insertion sort.
+
+### Components from glidesort which are not incorporated
+
+Some components/techniques from glidesort were intentionally not incorporated because they were considered too niche or have too much codesize overhead for a generic standard library sort:
+
+- **Merge splitting.** This allows arbitrarily small auxiliary buffers which was not necessary for `slice::sort` and enables interleaved merge loops which were not implemented for codesize reasons.
```

Comment:
I think we should settle one of either binary-size or codesize, and stay consistent with it.